### PR TITLE
updating plugin introduction

### PR DESCRIPTION
**Pull request by mhmohona**
## Explanation
Updated plugin PR adding more info about 
- What is a WasmEdge plugin?
- How can WasmEdge plugins be used?
- Benefits of using WasmEdge Plug-in System

## Related issue - 
#85 

Reopening PR #84 for passing DOC test

## What type of PR is this

/kind documentation

### CODE REVIEW

**Comment by alabulei1** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by adithyaakrishna** [comment]
@alabulei1 CI Workflows need your approval to run. Its a default feature of GH Actions. More Info: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

**Comment by alabulei1** [comment]
Hi  @mhmohona  Sorry for my delay. Could you please fix the conflicts?

File: package.json [hunk 1]
```diff
@@ -31,12 +31,14 @@
...
         "@docusaurus/core": "^2.4.0",
         "@docusaurus/preset-classic": "^2.4.0",
+        "@docusaurus/theme-mermaid": "^2.4.1",
         "@easyops-cn/docusaurus-search-local": "^0.35.0",
         "@mdx-js/react": "^1.6.22",
         "@node-rs/jieba": "^1.7.0",
         "classnames": "^2.3.2",
         "clsx": "^1.2.1",
         "dotenv": "^16.0.3",
+        "mermaid": "^10.2.3",
```
Comment (adithyaakrishna) [inline package.json:41]: We dont need this dependency as it is handled by `"@docusaurus/theme-mermaid": "^2.4.1",`

**Review by adithyaakrishna — COMMENTED** [review]
@mhmohona Could you please address the above comment?

**Review by alabulei1 — APPROVED** [review]

**Pull request closed without merging**